tactics/class_tactics.ml

```diff
@@ -764,7 +764,14 @@ module Search = struct
        This is an overapproximation. Evars could appear in this goal only
        and not any other *)
     let modes, poss = List.split poss in
-    let poss = List.flatten poss in
+    let poss = match poss with
```

Comment:
You should probably tweak `e_my_find_search` instead, because otherwise the immediate hints (in `e_trivial_resolve`) will not sort the hints across databases.